contracts/token/ERC721/ERC721.sol

```diff
@@ -165,7 +165,7 @@ contract ERC721 is ERC165, IERC721 {
      */
     function safeTransferFrom(address from, address to, uint256 tokenId, bytes memory _data) public {
         transferFrom(from, to, tokenId);
-        require(_checkOnERC721Received(from, to, tokenId, _data));
+        _safeTransferFrom(from, to, tokenId, _data);
```

Comment:
I think you may have misinterpreted @frangio's suggestion: the idea was for `_safeTransferFrom` to call both `_transferFrom` and `__checkOnERC721Received`.